manifests/custom_service.pp

```diff
@@ -45,30 +43,15 @@
   Enum['present','absent'] $ensure      = 'present',
 ) {
 
-  include firewalld::reload
-
-  # Service files may only contain alphanumeric characters and underscores.
-  # This is not documented, but has been experimentally confirmed.
-  $_safe_filename = firewalld::safe_filename($filename)
-
-  $_content = epp(
-    "${module_name}/service.xml.epp",
-    {
-      'short'       => $short,
-      'description' => $description,
-      'port'        => $port,
-      'module'      => $module,
-      'destination' => $destination,
-      'filename'    => $filename,
-      'config_dir'  => $config_dir,
-      'ensure'      => $ensure
-    }
-  )
+  $_args = delete_undef_values({
+    'description'      => $description,
+    'ports'            => $port,
+    'modules'          => $module,
+    'ipv4_destination' => $destination.dig('ipv4'),
+    'ipv6_destination' => $destination.dig('ipv6'),
+  })
 
-  file{ "${config_dir}/${_safe_filename}.xml":
-    ensure  => $ensure,
-    content => $_content,
-    mode    => '0644',
-    notify  => Class['firewalld::reload'],
+  firewalld_custom_service { $short:
```

Comment:
Should this use `$name` instead?

I have a manifest that currently includes
```puppet
firewalld::custom_service { 'foreman-smart-proxy':
    filename    => 'foreman-smart-proxy',
    short       => 'Foreman Smart Proxy',
    description => 'Smart Proxy is a free open source project that provides restful API to subsystems such as DNS, DHCP, etc, for higher level orchestration tools such as Foreman.',
    port        => [
      {
        'port'     => '8443',
        'protocol' => 'tcp',
      },
    ],
  }
```
After trying out this change, I now have 2 services defined.
```
Notice: /Stage[main]/Profile::Firewall::Foreman_proxy_rules/Firewalld::Custom_service[foreman-smart-proxy]/Firewalld_custom_service[Foreman Smart Proxy]/ensure: created
```
```
[root@foreman-pxe-server1 ~]# ls /etc/firewalld/services/foreman-smart-proxy.xml /etc/firewalld/services/Foreman_Smart_Proxy.xml 
/etc/firewalld/services/foreman-smart-proxy.xml  /etc/firewalld/services/Foreman_Smart_Proxy.xml
```

Reply:
Ah, I see. Let me check on this.

Reply:
@alexjfisher Ah...I think that the issue is that the module does not *purge* old entries. This is a consistent issue across the module and I think I have an issue out for it. The extraneous files won't do any harm (I think). The old method could create files that were invalid according to `firewalld` internals. The new method guarantees internal consistency.

Reply:
I guess I could add an `ensure => absent` for the old way of doing things as long as `$short != $name`. What do you think?

Reply:
The latest update should resolve this.